src/main/kotlin/com/gumi/moodle/model/User.kt

```diff
@@ -1,49 +1,31 @@
 package com.gumi.moodle.model
 
+import io.ktor.util.*
 import java.security.SecureRandom
 
-import java.security.spec.InvalidKeySpecException
-
-import javax.crypto.SecretKeyFactory
-
-import javax.crypto.spec.PBEKeySpec
-
 
 class User(
     var firstName: String,
     var lastName: String,
     var email: String,
-    var password: String,
-    var roles: List<Role> = listOf(Role.STUDENT),
-    var salt: ByteArray = ByteArray(0)
+    var password: String = "",
+    var salt: String = "",
+    var roles: List<Role> = listOf(Role.STUDENT)
 ) {
-    private val keyFactory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1")
+    private val digestFunction = getDigestFunction("SHA-256") { salt }
```

Comment:
SHA chyba nie powinno być stosowane do haseł, raczej coś typu BCRYPT, ale można zrobić jako osobny task w późniejszym sprincie.

Reply:
Jeśli mamy się później przenieść na OAutha to nie jest to chyba bardzo istotne.